src/aks-preview/azext_aks_preview/_helpers.py

```diff
@@ -374,3 +374,21 @@ def check_is_monitoring_addon_enabled(addons, instance):
     except Exception as ex:  # pylint: disable=broad-except
         logger.debug("failed to check monitoring addon enabled: %s", ex)
     return is_monitoring_addon_enabled
+
+
+def filter_hard_taints(node_initialization_taints: List[str]) -> List[str]:
```

Comment:
Can you help add some unit test for this function?

Reply:
Added in `test_helpers.py`, let me know if that looks good